kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/DisbursementVoucherDocumentPreRules.java

```diff
@@ -0,0 +1,125 @@
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.fp.businessobject.DisbursementVoucherSourceAccountingLine;
+import edu.arizona.kfs.fp.businessobject.DisbursementVoucherSourceAccountingLineExtension;
+import edu.arizona.kfs.fp.service.DisbursementVoucherInvoiceService;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSKeyConstants;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+
+public class DisbursementVoucherDocumentPreRules extends org.kuali.kfs.fp.document.validation.impl.DisbursementVoucherDocumentPreRules {
+
+    private static transient volatile DisbursementVoucherInvoiceService disbursementVoucherInvoiceService;
+    private static transient volatile ConfigurationService configurationService;
+
+    private static DisbursementVoucherInvoiceService getDisbursementVoucherInvoiceService() {
+        if (disbursementVoucherInvoiceService == null) {
+            disbursementVoucherInvoiceService = SpringContext.getBean(DisbursementVoucherInvoiceService.class);
+        }
+        return disbursementVoucherInvoiceService;
+    }
+
+    private static ConfigurationService getConfigurationService() {
+        if (configurationService == null) {
+            configurationService = SpringContext.getBean(ConfigurationService.class);
+        }
+        return configurationService;
+    }
+
+    @Override
+    public boolean doPrompts(Document document) {
+        boolean result = super.doPrompts(document);
+
+        result &= checkDisbursementVoucherInvoiceNumberRequired((DisbursementVoucherDocument) document);
+        result &= checkDisbursementVoucherInvoiceNumberDuplicate((DisbursementVoucherDocument) document);
+
+        return result;
+    }
+
+    @SuppressWarnings({ "deprecation", "unchecked" })
+    private boolean checkDisbursementVoucherInvoiceNumberRequired(DisbursementVoucherDocument document) {
+        boolean results = true;
+        DisbursementVoucherDocument disbursementVoucherDocument = (DisbursementVoucherDocument) document;
+        List<DisbursementVoucherSourceAccountingLine> accountingLines = disbursementVoucherDocument.getSourceAccountingLines();
+        for (DisbursementVoucherSourceAccountingLine accountingLine : accountingLines) {
+            DisbursementVoucherSourceAccountingLineExtension accountingLineExtension = accountingLine.getExtension();
+            if (StringUtils.isBlank(accountingLineExtension.getInvoiceNumber())) {
+                GlobalVariables.getMessageMap().putError(KFSPropertyConstants.SOURCE_ACCOUNTING_LINES, KFSKeyConstants.ERROR_REQUIRED, KFSConstants.INVOICE_NUMBER);
+                abortRulesCheck();
+                results = false;
+            }
+        }
+
+        return results;
+    }
+
+    @SuppressWarnings("deprecation")
+    private boolean checkDisbursementVoucherInvoiceNumberDuplicate(DisbursementVoucherDocument document) {
+
+        WorkflowDocument wfDoc = document.getDocumentHeader().getWorkflowDocument();

```

Comment:
This is where the validation for the original mod was performed, and when I was testing this, the warning question successfully appeared after clicking submit and approve. Removing this removes the warning question requested in the spec.